From ffe924ef060a9b9540a4dcd117e045eaefa62513 Mon Sep 17 00:00:00 2001
From: Alexander Graul <mail@agraul.de>
Date: Tue, 9 Mar 2021 13:46:03 +0100
Subject: [PATCH] 3002: Set distro requirement to oldest supported
 version (#327)

In the released Salt packages, python3-distro is taken from the
officially repositories on supported operating systems. The oldest
supported python3-distro version is 1.0.1 in Ubuntu18.04 universe and
Debian 9. FreeBSD is an exception and requires 1.3.0.
The mismatch between the version specified in requirements/base.txt and
what is actually used by the released packages can be confusing.

(cherry picked from commit 5c9c0ab9cdf2bf67bfdd259b53aa15297d1656ce)
(cherry picked from commit 0ff35358f79e9df8b06fb345fd79c1d22ed91179)

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>
---
 requirements/base.txt              | 2 +-
 requirements/static/pkg/freebsd.in | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/requirements/base.txt b/requirements/base.txt
index ffe4bc98f1..6af972bd1b 100644
--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -3,7 +3,7 @@ msgpack>=0.5,!=0.5.5
 PyYAML
 MarkupSafe
 requests>=1.0.0
-distro>=1.5
+distro>=1.0.1
 # Requirements for Tornado 4.5.3 (vendored as salt.ext.tornado)
 singledispatch==3.4.0.3; python_version < '3.4'
 # Required by Tornado to handle threads stuff.
diff --git a/requirements/static/pkg/freebsd.in b/requirements/static/pkg/freebsd.in
index 879a378822..7cfa3dcce8 100644
--- a/requirements/static/pkg/freebsd.in
+++ b/requirements/static/pkg/freebsd.in
@@ -8,3 +8,4 @@ python-dateutil>=2.8.0
 python-gnupg>=0.4.4
 setproctitle>=1.1.10
 timelib>=0.2.5
+distro>=1.3.0
-- 
2.30.1


